spiffworkflow-frontend/src/services/PermissionCacheService.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * There can be a lot of redundant requests for permissions (probably deps/contexts firing etc.)
+ * This service provides a cache to check for already-processed perms.
+ */
+import { PermissionsToCheck } from '../interfaces';
+
+/** Map makes sense: no prototype to hack, high perf, easily wiped. */
+const permissionCache = new Map<string, Array<string>>();
+
+const updatePermissionsCache = (permissionsToCheck: PermissionsToCheck) => {
+  if (permissionsToCheck) {
+    Object.keys(permissionsToCheck).forEach((path: string) => {
+      permissionCache.set(path, permissionsToCheck[path]);
+    });
+  }
+};
+
+/**
+ * TODO: This can lead to redundant permissions requests if the same individual permission
+ * is requested by different permission sets (any perm in a set not found will
+ * trigger a backend call for the whole set).
+ * This is erring on the side of caution for now, but a more robust individual
+ * checker might be useful.
+ */
+const findPermissionsInCache = (permissionsToCheck: PermissionsToCheck) => {
+  let foundAllPerms = true;
+  Object.keys(permissionsToCheck).forEach((path: string) => {
```

Comment:
Looks like this could use something like `find` or `some` to short circuit.

Reply:
I did originally short circuit (just returned false if anything was found) but we have a linter thing that does this:

![Screenshot 2024-03-10 at 8 21 01 AM](https://github.com/sartography/spiff-arena/assets/1213203/8d19a677-6fd2-48f4-abff-f581cc43c7fd)

(I've never seen a return value short circuit forbidden this way). 

Anyhow, I came up with something that should fit the bill (Array.every)